python/pyarrow/array.pxi

```diff
@@ -599,6 +599,20 @@ cdef class Array:
                                               self, &out))
         return wrap_array_output(out)
 
+    def to_numpy(self):
+        """
+        Convert to a NumPy array
+        """
+        if self.null_count:
+            raise NotImplementedError('NumPy array conversion is only '
+                                      'supported for arrays without nulls.')
+        if not is_primitive(self.type.id):
+            raise NotImplementedError('NumPy array conversion is only '
+                                      'supported for primitive types.')
+        buflist = self.buffers()
+        return np.frombuffer(buflist[-1], dtype=self.type.to_pandas_dtype())[
```

Comment:
Should we assert something about `len(buflist)` here?

Reply:
Done

Reply:
I was thinking `len(buflist) == 2` actually.